### ISSUE TITLE
Show hidden folders like .github in folder view

### ISSUE THREAD

**Issue by dar5hak** [elementary/code#1077]
## Problem

If I want to edit a GitHub Actions workflow file (located under `.github/workflows`), the only way to do it is to manually locate it in the **Open a file** chooser. This is because the `.github` folder does not show up in the folder view.

## Proposal

It would be helpful to show hidden folders like `.github` in the folder view itself when a project is open, so that a user can open it directly from there and track its Git status like other files.

Continuing the discussion from #936, this could be achieved by:

- showing all hidden folders in the folder view, possibly controlled by a setting, or
- showing specific hidden folders like `.github` by maintaining a list of useful ones, or
- showing all hidden folders that aren't git-ignored (see prior art below)

## Prior Art

VS Code, for instance, shows all directories in its folder view, but greys out the ones that are git-ignored, which does not include `.github`.

**Comment by dar5hak** [comment]
I encountered this while trying to add [bilelmoussaoui/flatpak-github-actions](https://github.com/bilelmoussaoui/flatpak-github-actions) to my workflow, which is something I believe many developers might want to do in the days leading up to Odin release.

**Comment by jeremypw** [comment]
It would be helpful get the view of @elementary/ux before starting to code as to the preferred solution.

**Comment by cassidyjames** [comment]
Yeah I think the original line of thought was to hide `.hidden` files/folders to clean up the sidebar, but since these files would be included in the Git repository regardless of being shown in the filesystem, perhaps we do just show them by default. Of course we could also add a "Show hidden files" toggle somewhere, but I am hesitant to add yet-another-setting.

Thoughts on just always showing hidden files in the sidebar, @danrabbit?

**Comment by danrabbit** [comment]
Yeah, I feel like it makes sense to always show hidden files and folders here since they'll likely be relevant for development

**Comment by igordsm** [comment]
On it.

**Comment by igordsm** [comment]
So, I got some progress on this today. To just show hidden items is easy: just changing the default value of `is_valid_directory` works OK. 

![Captura de tela de 2021-07-12 14-55-15@2x](https://user-images.githubusercontent.com/221446/125334010-440c1300-e321-11eb-99e4-37076a96d742.png)

Dimming icons is another story. The project view uses `Granite.Widgets.SourceList` and this uses `Glib.Icon` for rendering. As far as I understand its code, the function that actually draws the icon is [here](https://github.com/elementary/granite/blob/b1d725a6485b4c55bdcb660a9103aff785428b98/lib/Widgets/SourceList.vala#L2516). If I'm right, it does not allow for dimmed icons. 

Files does it differently and apparently implements its own IconRenderer ([here](https://github.com/elementary/files/blob/master/src/IconRenderer.vala)). I still have to understand how this class is used, but it seems that if more than one project implements "icon effects" maybe it should go into `Granite`? Any thoughts on that, @danrabbit ? I can send a PR for showing hidden files right away, but I'm waiting to see if I should bundle the visuals as well.

**Comment by jeremypw** [comment]
Deprioritizing folders is done by italicizing the name at the moment.  Not sure if Granite.Widgets.SourceList can be persuaded to dim the icons.  As you say, Files does this using a custom icon renderer which also handles such things colorizing and adding emblems.

**Comment by jeremypw** [comment]
Maybe you can dim the name?

**Comment by igordsm** [comment]
Dimming the name works (alpha=50%), but I'm not sure if the name of all children should be dimmed too. Right now it looks like this. In contrast, ignored files have alpha=75% and italics and apply this style to all children.

![Captura de tela de 2021-07-13 11-35-39](https://user-images.githubusercontent.com/221446/125471269-a3c40128-046b-40a1-a61d-d59a3bfb3263.png)

As comparison, VSCode does not dim the name or icons of hidden files and folders, but it does for ignored files. 

![Captura de tela de 2021-07-13 11-40-52](https://user-images.githubusercontent.com/221446/125472091-c580d812-ced5-4ca3-a0be-9dfea53d1011.png)

**Comment by jeremypw** [comment]
Needs UX team input on how to style (1) hidden and (2) ignored files.  Personally, I am not sure hidden files that are tracked need to be specially styled if they are always going to be displayed. Untracked files should be deprioritized.

**Comment by jeremypw** [comment]
@igordsm Do you have a draft PR that could be pushed?  Don't worry about styling hidden folders differently - all that matters is if they are tracked.

**Comment by igordsm** [comment]
@jeremypw Done. It's quite a simple fix, though
